Consolidated pull requests

sapohl opened the pull request:
needs-review: @vitillo 

This is to address the following bugs:
https://bugzilla.mozilla.org/show_bug.cgi?id=1284936
https://bugzilla.mozilla.org/show_bug.cgi?id=1284933

sapohl committed:
Break out orphaned users by usage hours

Break out orphaned users who haven't run a particular orphaned version
of Firefox for at least 2 hours.

sapohl committed:
Respect update_check_no_update_notify in update orphaning dashboard

We previously failed to check update_check_no_update_notify as a
possible state for orphaned users. This would mean that although
orphaned, users can’t see an update advertised and falsely believe that
no update is available.

vitillo commented:
Please rebase and create one commit per logical change.

sapohl commented:
If I followed instructions correctly, I'm hoping this is rebased properly now...

vitillo commented:
lgtm; let's merge this after rstrong r-plused

sapohl committed:
Read today's date from the environment param passed to the Spark job by Airflow

Also, remove use of camelCase for variable names

sapohl commented:
@vitillo : one more review request to get this job ready for Airflow

vitillo commented on reports/update-orphaning/Update orphaning analysis using longitudinal dataset.ipynb:
```diff
@@ -79,7 +80,12 @@
+    "#today = dt.date.today()\n",
+    "today = dt.datetime.strptime(today_env_str, \"%Y%m%d\")\n",
     "\n",
     "# Find the date of last Wednesday to get the proper 7 day range, and to find the last longitudinal dataset correctly.\n",
```
Note that you can't rely any longer on the longitudinal dataset being generated on a Wednesday and the variable names and the code should reflect that.

sapohl replied:
I think that's already addressed (you pointed this out to me before, i.e. that jobs might get backfilled on random days of the week). In the lines below, we find `lastWednesday` regardless of the particular day of the week that the job is run.

vitillo replied:
You don't need to find the last longitudinal dataset because you know that one exists precisely at the date passed in the environment variable. Airflow invokes this job right after generating a longitudinal dataset for that date.

sapohl commented:
@vitillo: updated and rebased

sapohl commented:
Rebased one more time to address https://bugzilla.mozilla.org/show_bug.cgi?id=1284933#c16.

@vitillo, if you're ok with the changes in https://github.com/mozilla-services/data-pipeline/pull/218/commits/a47fea0a41c1a500349cdce44bf9c55e806db5d6 we should be ready to merge now.

vitillo commented:
Is there any particular reason left why you have to find the last Wednesday? Couldn't you just subtract 15+ days from the passed `date` argument?

Also, the comment about "... to find the last longitudinal dataset correctly" shouldn't be there anymore I think.

sapohl commented:
Removed the comment and rebased, thanks for catching!

Yes, we still want to compare the same ranges, starting on the same weekday each time. Firefox usage varies based on days of the week, so we'd like to keep the comparisons 1-to-1. Also, the dashboard currently expects a new report every Wednesday.

vitillo commented:
lgtm
